### PR TITLE
Fixing data reload on stacked bar chart

### PR DESCRIPTION
**Pull request by Golodhros**
## Description

We are cleaning the old layers before redrawing the new data. This is still the non-optimal solution, but it will work for now.

## Motivation and Context

Related to https://github.com/eventbrite/britecharts/issues/291, this is part of the fixes for that issue.

## How Has This Been Tested?

Added tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/charts/stacked-bar.js [hunk 1]
```diff
@@ -411,46 +413,42 @@ define(function(require){
 
         /**
          * Draws the bars along the x axis
-         * @param  {D3Selection} bars Selection of bars
+         * @param  {D3Selection} layersSelection Selection of bars
          * @return {void}
          */
-        function drawHorizontalBars(series) {
+        function drawHorizontalBars(layersSelection) {
+            let layerJoin = layersSelection
```
Comment (rwholey-eb) [inline src/charts/stacked-bar.js]: new repo! any interest in switching to const when possible?

File: src/charts/stacked-bar.js [hunk 2]
```diff
@@ -473,44 +471,42 @@ define(function(require){
...
+        function drawVerticalBars(layersSelection) {
+            let layerJoin = layersSelection
+                .data(layers);
+
+            layerElements = layerJoin
                 .enter()
                   .append('g')
-                    .classed('layer', true)
                     .attr('fill', (({key}) => categoryColorMap[key]))
+                    .classed('layer', true);
```
Comment (rwholey-eb) [inline src/charts/stacked-bar.js]: layer seems generic, could this have conflicts with other charts down the road?

File: src/charts/stacked-bar.js [hunk 3]
```diff
@@ -473,44 +471,42 @@ define(function(require){
...
+            // Enter + Update
+            let bars = barJoin
                     .enter()
                       .append('rect')
                         .classed('bar', true)
                         .attr('x', (d) => xScale(d.data.key))
                         .attr('y', (d) => yScale(d[1]))
                         .attr('width', xScale.bandwidth )
-                        .attr('fill', (({data}) => categoryColorMap[data.stack+data.key])),context;
+                        .attr('fill', (({data}) => categoryColorMap[data.stack+data.key]));
```
Comment (rwholey-eb) [inline src/charts/stacked-bar.js]: could we do string interpolation here? `${data.stack}${data.key}`, for a second i thought you might be adding numbers

File: src/charts/stacked-bar.js (hunk 1 above)
Comment (Golodhros) [reply src/charts/stacked-bar.js]: It's not on my list of priorities, although I know Jaylum is more used to use const.

File: src/charts/stacked-bar.js (hunk 2 above)
Comment (Golodhros) [reply src/charts/stacked-bar.js]: Mmmm... good call. That could happen yes, I will keep an eye in this.

File: src/charts/stacked-bar.js (hunk 3 above)
Comment (Golodhros) [reply src/charts/stacked-bar.js]: Totally!

**Comment by jchen-eb** [comment]
LGTM, nice fix!

**Pull request merged**